src/index.ts

```diff
@@ -1,24 +1,97 @@
+import { expressMiddleware } from '@apollo/server/express4';
 import { addMocksToSchema } from '@graphql-tools/mock';
 import { makeExecutableSchema } from '@graphql-tools/schema';
+import { ApolloServerPluginDrainHttpServer } from '@apollo/server/plugin/drainHttpServer';
+import { ApolloServerPluginLandingPageLocalDefault } from '@apollo/server/plugin/landingPage/default';
+import { ApolloServerPluginLandingPageDisabled } from '@apollo/server/plugin/disabled';
 
-import { mysqlConfig } from './config';
-import { logger } from './logger';
-import { MysqlDataSource } from './datasources/mysqlDB';
-import { DMPHubAPI } from './datasources/dmphub-api';
-import { loggerPlugin } from './plugins/logger';
+import express from 'express';
+import http from 'http';
+import cors from 'cors';
 
 import { typeDefs } from './schema';
 import { resolvers } from './resolver';
 import { mocks } from './mocks';
+import { loggerPlugin } from './plugins/logger';
+
+import { logger } from './logger';
+import { mysqlConfig } from './config';
+import { DMPHubAPI } from './datasources/dmphub-api';
+import { MysqlDataSource } from './datasources/mysqlDB';
+
+// Required logic for integrating with Express
+const app = express();
+// Our httpServer handles incoming requests to our Express app.
+const httpServer = http.createServer(app);
+
+// Added this generic wrapper function to accomodate the fact that Typescript doesn't
+// allow a top level await
+async function startup(config, ): Promise<void> {
+  // TODO: We likely want to switch this up to validate the token and return the User in the JWT
+  function getTokenFromRequest(request) {
+    return request.headers?.authentication || '';
+  }
 
-// TODO: We likely want to switch this up to validate the token and return the User in the JWT
-function getTokenFromRequest(request) {
-  return request.headers?.authentication || '';
+  // Ensure we wait for our server to start
+  const server = new ApolloServer(config);
+  await server.start();
+
+  const { cache } = server;
+
+  // Healthcheck endpoint
+  app.get('/up', (_req, res) => {
+    server.executeOperation({ query: '{ __typename }' })
+          .then((data) => {
+            if (data.body.kind === 'single') {
+              if (data.body.singleResult.errors) {
+                const msgs = data.body.singleResult.errors.map((err) => err.message);
+                logger.error(`ERROR: Healthcheck - ${msgs.join(', ')}`);
+                res.status(400).send(JSON.stringify(data.body.singleResult.errors));
+              } else {
+                res.status(200).send(JSON.stringify(data.body.singleResult.data));
+              }
+            }
+          })
+          .catch((error) => {
+            logger.error(`ERROR: Healthcheck - ${error.message}`);
+            res.status(400).send(JSON.stringify(error));
+          });
+  });
+
+  // Set up our Express middleware to handle CORS, body parsing, and our expressMiddleware function.
+  app.use(
+    '/',
```

Comment:
@briri - eventually, when "app.use("/")" gets sort of larger, you could move out middleware functions into a separate locations, like 
"/middleware", and maybe just call them using something like:

```
app.use("/", (req, res, next) => {
   handleCors(req,res);
   extractToken(req, res);
   next();
});
```
or something like that.

Reply:
Yes, great idea. Better to do it now too while the code is relatively small. I moved the middleware into its own files and also moved the health check code into a separate file since we will be defining a few other non-graphQL pages to handle auth.

Reply:
@briri  Nice!